docs/user-guide/horizontal-pod-autoscaler.md

```diff
@@ -0,0 +1,126 @@
+
+The autoscaler is implemented as a Kubernetes API resource and a controller.
+The resource describes behavior of the controller.
+The controller periodically adjusts the number of replicas in replication controller or deployment
+to match the observed average CPU utilization to the target specified by user.
+
+
+## How does Horizontal Pod Autoscaler work?
+
+![Horizontal Pod Autoscaler diagram](horizontal-pod-autoscaler.png)
+
+The autoscaler is implemented as a control loop.
+It periodically queries CPU utilization of pods it is responsible for.
+(The period of the autoscaler is controlled by `--horizontal-pod-autoscaler-sync-period` flag of controller manager.
+The default value is 30 seconds).
+Then, it compares the arithmetic mean of the pods' CPU utilization with the target and adjust the number of replicas if needed.
+
+CPU utilization is the recent CPU usage of a pod divided by the sum of CPU requested by the pod's containers.
+Please note that if some of the pod's containers do not have CPU request set,
+CPU utilization for the pod will not be defined and the autoscaler will not take any action.
+Further details of the autoscaling algorithm are given [here](../proposals/horizontal-pod-autoscaler.md#autoscaling-algorithm).
+
+Autoscaler accesses corresponding replication controller or deployment by scale sub-resource.
+Scale is an interface which allows to dynamically set the number of replicas and to learn the current state of them.
+More details on scale sub-resource can be found [here](../proposals/horizontal-pod-autoscaler.md#scale-subresource).
+
+
+## API Object
+
+Horizontal pod autoscaler is a top-level resource in the Kubernetes REST API (currently in [beta](../api.md#api-versioning)).
+More details about the API object can be found at
+[HorizontalPodAutoscaler Object](../proposals/horizontal-pod-autoscaler.md#horizontalpodautoscaler-object).
+
+## Support for horizontal pod autoscaler in kubectl
+
+Horizontal pod autoscaler, like every API resource, is supported in a standard way by `kubectl`.
+We can create a new autoscaler using `kubectl create` command.
+We can list autoscalers by `kubectl get hpa` and get detail description by `kubectl describe hpa`.
+Finally, we can delete an autoscaler using `kubectl delete hpa`.
+
+In addition, there is a special `kubectl autoscale` command that allows for easy creation of horizontal pod autoscaler.
+For instance, executing `kubectl autoscale rc foo --min=2 --max=5 --cpu-percent=80`
+will create an autoscaler for replication controller *foo*, with target CPU utilization set to `80%`
+and the number of replicas between 2 and 5.
+The detailed documentation of `kubectl autoscale` can be found [here](kubectl/kubectl_autoscale.md).
+
+
+## Autoscaling during rolling update
+
+The question arises how autoscaler interacts with an ongoing rolling update.
+If the rolling update is done by a deployment and horizontal pod autoscaler is bind to the deployment, they interact correctly.
```

Comment:
@ncdc @DirectXMan12 @ironcladlou @smarterclayton - I am not sure we discussed this scenario with using HPA against a OpenShift deployment.  We need to validate this scenario